great_expectations/rule_based_profiler/domain_builder/categorical_column_domain_builder.py

```diff
@@ -0,0 +1,247 @@
+            )
+
+        first_batch_id: str = batch_ids[0]
+
+        table_column_names: List[str] = validator.get_metric(
+            metric=MetricConfiguration(
+                metric_name="table.columns",
+                metric_domain_kwargs={
+                    "batch_id": first_batch_id,
+                },
+                metric_value_kwargs=None,
+                metric_dependencies=None,
+            )
+        )
+
+        if self._exclude_columns is not None:
+            table_column_names = [
+                colname
+                for colname in table_column_names
+                if colname not in self._exclude_columns
+            ]
+
+        return table_column_names
+
+    def _generate_metric_configurations_to_check_cardinality(
+        self,
+        batch_ids: List[str],
+        column_names: List[str],
+    ) -> List[Dict[str, List[MetricConfiguration]]]:
+        """Generate metric configurations used to compute metrics for checking cardinality.
+
+        Args:
+            batch_ids: List of batch_ids used to create metric configurations.
+            column_names: List of column names used to create metric configurations.
+
+        Returns:
+            List of dicts of the form [{column_name: List[MetricConfiguration]},...]
+        """
+
+        limit_mode: CardinalityLimitMode = self._cardinality_checker.limit_mode
+
+        metric_name: str = limit_mode.related_metric_name
+
+        metric_configurations: List[Dict[str, List[MetricConfiguration]]] = []
+
+        for column_name in column_names:
+            metric_configurations.append(
+                {
+                    column_name: [
+                        MetricConfiguration(
+                            metric_name=metric_name,
+                            metric_domain_kwargs={
+                                "batch_id": batch_id,
+                                "column": column_name,
+                            },
+                            metric_value_kwargs=None,
+                            metric_dependencies=None,
+                        )
+                        for batch_id in batch_ids
+                    ]
+                }
+            )
+
+        return metric_configurations
+
+    def _columns_meeting_cardinality_limit(
+        self,
+        validator: Validator,
+        table_column_names: List[str],
+        metrics_for_cardinality_check: List[Dict[str, List[MetricConfiguration]]],
+    ) -> List[str]:
+        """Compute cardinality and return column names meeting cardinality limit.
+
+        Args:
+            validator: Validator used to compute column cardinality.
+            table_column_names: column names to verify cardinality.
+            metrics_for_cardinality_check: metric configurations used to
+                compute cardinality.
+
+        Returns:
+            List of column names meeting cardinality.
+        """
+        candidate_column_names: List[str] = table_column_names.copy()
```

Comment:
@anthonyburdi Why not use `copy.deepcopy()`?  Is this a different object not amenable to `copy.deepcopy()`?..  Thanks!